cmd/ethrex_l2/src/commands/stack.rs

```diff
@@ -164,6 +191,83 @@ impl Command {
                     println!("Aborted.");
                 }
             }
+            Command::BlobsSaver {
+                l1_eth_rpc,
+                l1_beacon_rpc,
+                contract_address,
+                data_dir,
+            } => {
+                create_dir_all(data_dir.clone())?;
+
+                let eth_client = EthClient::new(l1_eth_rpc.as_str());
+                let beacon_client = BeaconClient::new(l1_beacon_rpc);
+
+                // Keep delay for finality
+                let mut current_block = eth_client.get_block_number().await? - U256::from(64);
+                let event_signature = keccak("BlockCommitted(bytes32)");
+
+                loop {
+                    // Wait for a block
+                    sleep(Duration::from_secs(12));
+
+                    let logs = eth_client
+                        .get_logs(
+                            current_block,
+                            current_block,
+                            contract_address,
+                            event_signature,
+                        )
+                        .await?;
+
+                    if !logs.is_empty() {
+                        // Get parent beacon block root hash from block
+                        let block = eth_client
+                            .get_block_by_number(BlockByNumber::Number(current_block.as_u64()))
+                            .await?;
+                        let parent_beacon_hash = block
+                            .header
+                            .parent_beacon_block_root
+                            .ok_or_eyre("Unknown parent beacon root")?;
+
+                        // Get block slot from parent beacon block
+                        let parent_beacon_block =
+                            beacon_client.get_block_by_hash(parent_beacon_hash).await?;
+                        let target_slot = parent_beacon_block.message.slot + 1;
+
+                        // Get blobs from block's slot
+                        let mut block_blobs = beacon_client.get_blobs_by_slot(target_slot).await?;
+
+                        // Get versioned hashes from transactions
+                        let mut l2_blob_hashes = vec![];
+                        for log in logs {
+                            let tx = eth_client
+                                .get_transaction_by_hash(log.transaction_hash)
+                                .await?
+                                .ok_or_eyre(format!(
+                                    "Transaction {:#x} not found",
+                                    log.transaction_hash
+                                ))?;
+                            l2_blob_hashes.extend(tx.blob_versioned_hashes.ok_or_eyre(format!(
+                                "Blobs not found in transaction {:#x}",
+                                log.transaction_hash
+                            ))?);
+                        }
```

Comment:
Could this be moved?
```suggestion
                        // Get versioned hashes from transactions
                        let mut l2_blob_hashes = vec![];
                        for log in logs {
                            let tx = eth_client
                                .get_transaction_by_hash(log.transaction_hash)
                                .await?
                                .ok_or_eyre(format!(
                                    "Transaction {:#x} not found",
                                    log.transaction_hash
                                ))?;
                            l2_blob_hashes.extend(tx.blob_versioned_hashes.ok_or_eyre(format!(
                                "Blobs not found in transaction {:#x}",
                                log.transaction_hash
                            ))?);
                        }
                        
                        // Get blobs from block's slot
                        let mut block_blobs = beacon_client.get_blobs_by_slot(target_slot).await?;
```